src/main/java/amazon/HomePage.java

```diff
@@ -71,11 +72,13 @@ public void chooseType(String type) {
         WebElement chooseCategory = driver.findElement(By.xpath("//a[text()='" + type + "']"));
         chooseCategory.click();
     }
+    public void dealsOfTodayButtonClick(){
+        dealsOfTodayButton.click();
+    }
 
 
+    @Override
     public void waitForPageLoad() {
-        new WebDriverWait(driver, Duration.ofSeconds(15)).until(ExpectedConditions.elementToBeClickable(allButtonClick));
-
-
+        new WebDriverWait(driver,Duration.ofSeconds(20)).until(ExpectedConditions.visibilityOf(dealsOfTodayButton));
```

Comment:
Why you changed it to wait for Deals of today button? So every time you have another flow to test, you should change the waitForPageLoad method? And what if we ran the test suite, I mean all the test sequentially, what should your waitForPageLoad method wait for?
Write one, normal waitForPageLoad method, after which execution you can be sure that the page has been loaded.